lib/graphql/stitching/util.rb

```diff
@@ -50,6 +50,14 @@ def self.get_possible_types(schema, parent_type)
       def self.is_leaf_type?(type)
         type.kind.scalar? || type.kind.enum?
       end
+
+      def self.get_field_type(supergraph, parent_type, node)
```

Comment:
This should accept a plain `GraphQL::Schema` rather than Supergraph. The argument can be passed in as `supergraph.schema`. Util wants to remain purely focused on GraphQL primitives. The name might also be more explicit as `get_named_type_for_field_node`.